python/ray/rllib/agents/agent.py

```diff
@@ -316,7 +318,13 @@ def train(self):
             logger.debug("synchronized filters: {}".format(
                 self.local_evaluator.filters))
 
-        return Trainable.train(self)
+        result = Trainable.train(self)
+        if self.config["callbacks"].get("on_train_step"):
+            self.config["callbacks"]["on_train_step"]({
+                "agent": self,
+                "result": result,
```

Comment:
.copy()?

Reply:
(or deepcopy)

Reply:
I think the advantage here is the user can mutate it if they want.